microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -47,6 +55,164 @@ class VmSnapsotLinuxBVTExtension(TestSuite):
+                        "FileSystemConsistent or ApplicationConsistent"
+                    )
+            else:
+                log.info(f"rp status is {response.provisioning_state}")
+                attempts += 1
+                time.sleep(2)
+        raise ValueError(
+            "Restore point provisioning status not Succeeded "
+            "after multiple attempts."
+        )
+
+    def find_extension_dir(self, node: Node) -> str:
+        startpath = PurePath("/var/lib/waagent/")
+        namepattern = "Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.*"
+        sudo = True
+        find_tool = node.tools[Find]
+        file_list = find_tool.find_files(
+            start_path=startpath, name_pattern=namepattern, sudo=sudo
+        )
+        extension_directory = ""
+        if len(file_list) == 1:
+            extension_directory = file_list[0]
+        return extension_directory
+
+    def copy_to_node(self, node: Node, filename: str) -> None:
+        f_name = node.working_path / filename
+        temp_file = PurePosixPath(filename)
+        file_path = PurePath(
+            os.path.join((os.path.dirname(__file__)), "scripts", temp_file)
+        )
+        node.shell.copy(file_path, f_name)
+
+    @TestCaseMetadata(
+        description="""
+        Runs the custom script extension and verifies it execution on the
+        remote machine.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension],
+        ),
+    )
+    def verify_exclude_disk(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        result: TestResult,
+    ) -> None:
+        # Any extension will do, use CustomScript for convenience.
+        # Copy the local files into the VM.
+        # Move the files to the extension directory
+        # Give the execution permissions to the files
+        # Run the files
+        # Validate the results
+        extension_dir = self.find_extension_dir(node)
+        if extension_dir == "":
+            self._verify_vmsnapshot_extension(log, node, environment)
+            trial = 0
+            while trial < 3:
+                time.sleep(2)
+                extension_dir = self.find_extension_dir(node)
+                if extension_dir == "":
+                    trial = trial + 1
+                else:
+                    break
+
+        # installing all the required packages
+        posix_os: Posix = cast(Posix, node.os)
+        posix_os.install_packages("python2.7")
+        posix_os.install_packages("pip")
+        url = "https://bootstrap.pypa.io/pip/2.7/get-pip.py"
+        args = "-o get-pip.py"
+        curl = Curl(node)
+        curl.fetch(url=url, arg=args, shell=True, execute_arg="", sudo=True)
+        node.execute(cmd="python2.7 get-pip.py", shell=True, sudo=True)
+        node.execute(cmd="python2.7 -m pip --version")
+        node.execute(cmd="python2.7 -m pip install mock", sudo=True)
```

Comment:
Can you please add more details about why these commands need to be run?

Reply:
Can you use Python3, and use the Python tool in LISA? The Python tool of LISA can ensure Python is installed in different distros.